### PR TITLE
Modification of names used for credentials

### PR DESCRIPTION
**Pull request by sbezverk**
This PR makes credentials related fields more intuitive and less confusing.
Closes: #140

### CODE REVIEW

**Comment by jdef** [comment]
Thanks for the PR. Related to #140 ?

Also, please don't change generated files manually. Instead the root-level spec.md should be changed, and then run `make` to regenerate downstream files.

**Comment by sbezverk** [comment]
@jdef You are right, it is related to #140 I just make it a little more intuitive by using controller_credentials and node_credentials instead of just credentials as suggested.

**Comment by sbezverk** [comment]
@jdef @saad-ali  Any hints way check is failing?

**Comment by jdef** [comment]
@sbezverk I expect to see 3 files changed, 2 of them are for generated files (csi.proto and the golang bindings (pb.go file)) and 1 file is manually altered (the changes made to spec.md). After making changes to spec.md, run `make` and it will generate the other files for you.

File: spec.md [hunk 1]
```diff
@@ -1074,18 +1074,18 @@ message NodePublishVolumeRequest {
...
   // '_' or '.'. Each value MUST contain a valid string. An SP MAY
   // choose to accept binary (non-string) data by using a binary-to-text
   // encoding scheme, like base64. An SP SHALL advertise the
   // requirements for credentials in documentation. COs SHALL permit
-  // users to pass through the required credentials. This information is
+  // passing through the required credentials. This information is
   // sensitive and MUST be treated as such (not logged, etc.) by the CO.
   // This field is OPTIONAL.
-  map<string, string> user_credentials = 7;
+  map<string, string> node_credentials = 7;
```
Comment (jieyu) [inline spec.md]: As discussed in #140, `NodePublishVolume` might be called one per workload by a CO. Calling it `node_credentials` makes it sound like the credentials are for the entire node.

File: spec.md [hunk 2]
```diff
@@ -478,18 +478,18 @@ message CreateVolumeRequest {
...
   // '_' or '.'. Each value MUST contain a valid string. An SP MAY
   // choose to accept binary (non-string) data by using a binary-to-text
   // encoding scheme, like base64. An SP SHALL advertise the
   // requirements for credentials in documentation. COs SHALL permit
-  // users to pass through the required credentials. This information is
+  // passing through the required credentials. This information is
   // sensitive and MUST be treated as such (not logged, etc.) by the CO.
   // This field is OPTIONAL.
-  map<string, string> user_credentials = 6;
+  map<string, string> controller_credentials = 6;
```
Comment (jieyu) [inline spec.md]: It's possible that the credentials used for `CreateVolume`/`DestroyVolume` are different than those used for `ControllerPublishVolume`/`ControllerUnpublishVolume`. Calling them `controller_credentials` makes it sounds like they should be the same. I think we either follow the suggestion in #140, or just name it as "credentials".

File: spec.md (hunk 1 above)
Comment (sbezverk) [reply spec.md]: @jdef before you had user_credentials which made sound like the same credential is used for everything.
controller_credential and node_credential are per function, the content of it depends on the usage of corresponding function controller or node. Making them even more granular is an option but imho is an overkill.

File: spec.md (hunk 2 above)
Comment (saad-ali) [reply spec.md]: I agree, let's go with just `credentials` or `create_credentials`, `delete_credentials`, `controller_publish_credentials`, etc.

**Comment by sbezverk** [comment]
@saad-ali  done changes with slight modification, since controller and node has publish/unpublish calls, I suggest to use the following model: controller_create_credentials, controller_publish_credentials, node_publish_credentials etc. In this case there is no confusion which credentials are used for what purpose.

**Comment by saad-ali** [comment]
This LGTM

@jieyu @jdef ?

**Comment by sbezverk** [comment]
@saad-ali Could you mark your review as completed then? thanks a lot for reviewing.

File: spec.md [hunk 3]
```diff
@@ -1137,26 +1137,27 @@ message NodeUnpublishVolumeRequest {
...
+  map<string, string> node_unpublish_credentials = 4;
 }
 
 message NodeUnpublishVolumeResponse {}
 ```
 
 ##### NodeUnpublishVolume Errors
 
-If the plugin is unable to complete the NodeUnpublishVolume call successfully, it MUST return a non-ok gRPC code in the gRPC status.
+If the plugin is unable to
```
Comment (jdef) [inline spec.md]: please undo this seemingly unintentional change

Comment (sbezverk) [reply spec.md]: thank you for catching it, fixed.

**Pull request merged**